client/src/js/services/FiscalService.js

```diff
@@ -21,5 +21,84 @@ function FiscalService(PrototypeApiService) {
   // the service URL
   service.url = '/fiscal/';
 
+  service.read = read;
+  service.fiscalYearDate = fiscalYearDate;
+
+  service.create = create;
+  service.update = update;
+  service.delete = del;
+
+  /**
+  * @desc Get an id (optionnal) and return back a list of Fiscal Years or a Fiscal year
+  * @param {Integer} id, the id of the Fiscal Year (optionnal) 
+  * @return {object} a promise object, with the response.body inside.
+  * @example
+  * service.read()
+  * .then(function (Fiscal Year){
+  *   your code here
+  *  });
+  **/
+  function read(id, params) {
```

Comment:
These methods already exist on the prototype service, so they are not necessary to create here.  See #421 for more information.

How does this work?  The PrototypeApiService defines common methods, like `read()` to work on the URL you pass into it.  You can inherit the methods using `angular.extend()`, like this:
```js
// this will extend service with the read(), create(), update(), and delete() methods.
angular.extend(service, PrototypeApiService, { url: '/fiscal' });

```

See the [VoucherService](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/VoucherService.js#L19) for one implementation.

The only new method you need to add is the `fiscalYearDate(params)` method, I think.